src/table/table-role/grid-navigation-context.tsx

```diff
@@ -0,0 +1,43 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
The grid navigation is a feature of the table and those interactive elements that can be rendered in the content have a dependency on the table as result. I would move it to a shared folder in case there is a indeed an opportunity to reuse the logic but it is hard to make this call just yet as it is not clear which parts can be shared. Say, the navigation commands in a different component with a similar focus overtake can be different.

Reply:
Why not now?

Reply:
Actually, I was thinking more about the grid navigation feature which is coupled to the table as of yet. The context is less coupled though. What it actually controls is focusability of the elements. When present, the elements underneath can be instructed to "mute" user focus so it can be some generic UserFocusController context or alike. However, we might want to make alterations to some elements like select/multiselect when used within keyboard navigation and this can in theory be different depending on what kind of navigation is that.

What do you think if I move the context to under /internal and rename as such:
```
interface SingleTabStopNavigationContext {
  focusTarget: null | Element;
  navigationActive: boolean;
}
```

This can be extended in the future with more properties to carry more information the components can use to alter their behaviour if needed.

Reply:
Everything that does not make every component importing from `../table`, works